src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -641,322 +635,146 @@ private async Task TryNotifyClientErrorAsync(IClientProxy client, string error,
-                _circuitHandlerFailed(
+                CircuitHandlerFailed(
                     logger,
                     handler.GetType(),
                     handlerMethod,
                     exception.Message,
                     exception);
             }
 
-            public static void CircuitUnhandledException(ILogger logger, CircuitId circuitId, Exception exception) => _circuitUnhandledException(logger, circuitId, exception);
-            public static void CircuitTransmitErrorFailed(ILogger logger, CircuitId circuitId, Exception exception) => _circuitTransmitErrorFailed(logger, circuitId, exception);
-            public static void EndInvokeDispatchException(ILogger logger, Exception ex) => _endInvokeDispatchException(logger, ex);
-            public static void EndInvokeJSFailed(ILogger logger, long asyncHandle, string arguments) => _endInvokeJSFailed(logger, asyncHandle, arguments, null);
-            public static void EndInvokeJSSucceeded(ILogger logger, long asyncCall) => _endInvokeJSSucceeded(logger, asyncCall, null);
-            public static void ReceiveByteArraySuccess(ILogger logger, long id) => _receiveByteArraySuccess(logger, id, null);
-            public static void ReceiveByteArrayException(ILogger logger, long id, Exception ex) => _receiveByteArrayException(logger, id, ex);
-            public static void ReceiveJSDataChunkException(ILogger logger, long streamId, Exception ex) => _receiveJSDataChunkException(logger, streamId, ex);
-            public static void SendDotNetStreamException(ILogger logger, long streamId, Exception ex) => _sendDotNetStreamException(logger, streamId, ex);
-            public static void DispatchEventFailedToParseEventData(ILogger logger, Exception ex) => _dispatchEventFailedToParseEventData(logger, ex);
-            public static void DispatchEventFailedToDispatchEvent(ILogger logger, string eventHandlerId, Exception ex) => _dispatchEventFailedToDispatchEvent(logger, eventHandlerId ?? "", ex);
+            [LoggerMessage(111, LogLevel.Error, "Unhandled exception in circuit '{CircuitId}'.", EventName = "CircuitUnhandledException")]
+            public static partial void CircuitUnhandledException(ILogger logger, CircuitId circuitId, Exception exception);
+
+            [LoggerMessage(114, LogLevel.Debug, "Failed to transmit exception to client in circuit '{CircuitId}'.", EventName = "CircuitTransmitErrorFailed")]
+            public static partial void CircuitTransmitErrorFailed(ILogger logger, CircuitId circuitId, Exception exception);
+
+            [LoggerMessage(204, LogLevel.Debug, "There was an error invoking 'Microsoft.JSInterop.DotNetDispatcher.EndInvoke'.", EventName = "EndInvokeDispatchException")]
+            public static partial void EndInvokeDispatchException(ILogger logger, Exception ex);
+
+            [LoggerMessage(205, LogLevel.Debug, "The JS interop call with callback id '{AsyncCall}' with arguments {Arguments}.", EventName = "EndInvokeJSFailed")]
```

Comment:
This log statement changed. It used to previously say:

> "The JS interop call with callback id '{AsyncCall}' failed with error '{Error}'."

but Error is not value that appears in the code. The calling code does say it should be safe to include the Arguments: in the log message, so this was probably a wrong log statement all along 

https://github.com/dotnet/aspnetcore/blob/832ea98014f9a49fceecf6b34e8c8b4c16b48ff5/src/Components/Server/src/Circuits/CircuitHost.cs#L371